examples/templates/suite_com/test_com.py

```diff
@@ -59,20 +59,44 @@ def test_run(self):
         logging.info(
             f"--------------- RUN: {self.test_suite_id}, {self.test_case_id} ---------------"
         )
-        # send 20 requests over the connected channel and check if the
-        # command was successfully sent
+
+        # send 20 requests (without context manager) over the connected channel
+        # and check if the command was successfully sent
         for _ in range(20):
             req = b"\x02\x04\x06"
             logging.info(f"send request {req} over {com_aux.name}")
             state = com_aux.send_message(req)
             logging.info(f"request excecution state: {state}")
             self.assertEqual(state, True)
 
-        # get the 20 first received messages
-        for _ in range(20):
-            response = com_aux.receive_message()
-            logging.info(f"received data {response}")
-            self.assertEqual(response, b"\x02\x04\x06")
+        # check messages have not been queued in as
+        # context manager wasn't used
+        self.assertTrue(com_aux.queue_out.empty())
```

Comment:
From the test above, there is no element known that will populate the queue_out. Am I missing something?

Reply:
To populate the queue send_message need to be in the ctx manager, otherwise you can not retrieve any messages